lib/carrierwave/storage/fog.rb

```diff
@@ -1,6 +1,10 @@
 # encoding: utf-8
 
-require "fog" unless defined?(::Fog)
+begin
+  require "fog/core"
+rescue LoadError
+  require "fog" unless defined?(::Fog)
```

Comment:
At this point I'm not sure if we're saying the same thing :)

Your current code is correct for this file. We need both requires to handle 3 scenarios:
1. You're using an up-to-date version of Fog that has fog/core
2. You're using an older version of Fog that requires you to load the whole thing
3. Fog is not included in the gem bundle and cannot be required, which should `raise` since either 1 or 2 is required to continue.